api/application_submissions/index.js

```diff
@@ -31,24 +31,18 @@ router.post('/', async (req, res) => {
   }
 });
 
-router.put('/:id', async (req, res) => {
-  const { id } = req.params;
+router.put('/', async (req, res) => {
+  const ids = req.query.id;
   const { decision } = req.body;
 
   try {
-    let applicationSubmission = await ApplicationSubmission.findById(id);
-    if(applicationSubmission) {
-      if(decision == 'ACCEPTED') {
-        const { skilled_impact_request_id } = applicationSubmission;
-        applicationSubmission = await ApplicationSubmission.acceptAndDenyAllOthers(id, skilled_impact_request_id);
-      }
-      else {
-        applicationSubmission = await ApplicationSubmission.update(id, decision);
-      }
-      res.status(200).json({ applicationSubmission, message: 'Submission updated in database' });
+    let applicationSubmissions = await ApplicationSubmission.findAllByIds(ids);
+    if(applicationSubmissions.length > 0) {
+      applicationSubmissions = await ApplicationSubmission.updateAll(ids, decision);
+      res.status(200).json({ applicationSubmissions, message: 'Submissions updated in database' });
     }
     else {
-      res.status(404).json({ message: 'Submission not found in database' });
+      res.status(404).json({ message: 'Submissions not found in database' });
```

Comment:
I think `ApplicationSubmissions.updateAll` returns all the rows that're changed (or at least a count, but you should double check this), you don't do anything with the `findAllByIds` data, and the `updateAll` only changes applications with valid IDs, so you can just call `updateAll` and check the return value for your conditional and maybe even make the error message even more specific by adding the number of IDs that weren't found, e.g. if 10 are passed in but 8 are updated, say 2 not found.